main.ts

```diff
@@ -1,135 +1,110 @@
-				.expandAndWrap("`", "`", editor),
-		});
-		this.addCommand({
-			id: "smarter-highlight",
-			name: "Smarter Highlight",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("==", "==", editor),
-		});
-		this.addCommand({
-			id: "smarter-strikethrough",
-			name: "Smarter Strikethrough",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("~~", "~~", editor),
-		});
-		this.addCommand({
-			id: "smarter-wikilink",
-			name: "Smarter wikilink (internal link)",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[[", "]]", editor),
-		});
-		this.addCommand({
-			id: "smarter-md-link",
-			name: "Smarter Markdown Link",
-			editorCallback: (editor: Editor, view: MarkdownView) => this
-				.expandAndWrap("[", "]()", editor),
+		COMMANDS.forEach((command) => {
+			const { id, name, before, after } = command;
+			this.addCommand({
+				id,
+				name,
+				editorCallback: (editor) =>
+					this.expandAndWrap(before, after, editor),
+			});
 		});
 
-		console.log ("Smarter MD Hotkeys loaded.");
+		console.log("Smarter MD Hotkeys loaded.");
 	}
 
 	async onunload() {
-		console.log ("Smarter MD Hotkeys unloaded.");
+		console.log("Smarter MD Hotkeys unloaded.");
 	}
 
 	expandAndWrap(beforeStr: string, afterStr: string, editor: Editor): void {
+		// Arrow functions are simpler
+		// No need to specify the return-type, Typescript can infer it
+		const offToPos = (offset: number) => editor.offsetToPos(offset);
 
-		function offToPos(offset: number): EditorPosition {
-			return editor.offsetToPos(offset);
-		}
-		function WordUnderCursor (ep: EditorPosition): Range{
+		// Return type is infered
+		function WordUnderCursor(ep: EditorPosition) {
 			// https://codemirror.net/doc/manual.html#api_selection
-			if (editor.cm?.findWordAt) return editor.cm.findWordAt(ep); //CM5
-			else if (editor.cm?.state.wordAt) return editor.cm.state.wordAt(editor.posToOffset(ep)); //CM6
+			if (editor.cm?.findWordAt) return editor.cm.findWordAt(ep);
+			//CM5
+			else if (editor.cm?.state.wordAt)
+				return editor.cm.state.wordAt(editor.posToOffset(ep)); //CM6
 		}
-		function trimSelection (trimBefArray: string[], trimAftArray: string[]): void{
+
+		function trimSelection(
+			trimBefArray: string[],
+			trimAftArray: string[]
+		): void {
 			let selection = editor.getSelection();
 			let sp = editor.posToOffset(editor.getCursor("from"));
 
 			//before
 			let trimFinished = false;
-			while (!trimFinished){
+			while (!trimFinished) {
 				let cleanCount = 0;
-				trimBefArray.forEach(str =>{
+				trimBefArray.forEach((str) => {
 					if (selection.startsWith(str)) {
-						selection = selection.slice (str.length);
+						selection = selection.slice(str.length);
 						sp += str.length;
 					} else cleanCount++;
 				});
-				if (cleanCount == trimBefArray.length || selection.length == 0) trimFinished = true;
+				// `a.length === 0` is the same as `!a.length`

```

Comment:
👀

Reply:
This works because `0` is treated as `false` and anything `1` or higher is treated as `true`.
We say `0` is "falsey" and `1` is "truthy".
So it's not exactly the same as `false`, but and `if` statement will treat them the same.

So `if(a.length === 0)` is the same as `if(!a.length)`